packages/components/src/CollapsiblePanel/CollapsiblePanel.component.js

```diff
@@ -84,107 +102,178 @@ renderHeaderItem.propTypes = PropTypes.oneOfType([
-			tooltipPlacement: 'right',
-			hideLabel: true,
-			link: true,
-		}
-	],
-	content: [
-		{
-			label: 'Content1',
-			description: 'Description1',
-		},
-		{
-			label: 'Content2',
-			description: 'Description2',
-		}
-
-	]
-};
- <CollapsiblePanel {...props} />
+ *
+ * @param header required, an array gathering the header elements except the caret
+ * @param onSelect optional, on header click callback function
+ * @param onToggle optional, on caret click callback function
+ * @param expanded optional, defines if the panel should be expanded or not
+ * @param selected optional, defines if the panel is selected or not
+ * @param theme optional, defines the theme of the collapsible, there is a default theme
+ * @param content optional, defines the content of the panel's body:
+ * if content is an array a key value content list is rendered otherwise it is a textual content
+ *
+ * @example
+ * <CollapsiblePanel {...props} />
  */
-function CollapsiblePanel({ header, content }) {
-	const hasContent = content && content.length;
-	const headerItems = renderHeader({ header, caret: hasContent });
+function CollapsiblePanel({ header, content, onSelect, onToggle, selected, expanded, theme }) {
+	const headerItems = renderHeader(header, content, onSelect, onToggle);
+	const className = classNames(
+		'panel panel-default',
+		css['tc-collapsible-panel'],
+		{
+			[css['default-panel']]: !theme,
+			[css['descriptive-panel']]: theme,
+			[css.selected]: selected,
+			[css.open]: expanded,
+		});
 
-	if (hasContent) {
-		return (
-			<Panel className={css['tc-collapsible-panel']} collapsible header={headerItems}>
-				{content.map(
-					(item, index) => ((
-						<div key={index} className={css.content}>
-							<div className={css.label}><Label>{item.label}</Label></div>
-							<div className={css.description}>{item.description}</div>
-						</div>
-						)),
-					)}
-			</Panel>
-		);
+	let children = null;
+	if (content) {
+		children = Array.isArray(content) ? getKeyValueContent(content) : getTextualContent(content);
 	}
-	return (<Panel className={`tc-collapsible-panel ${css['tc-panel']}`} header={headerItems} />);
+	return (
+		<div className={className}>
+			<div className={classNames(css['panel-heading'], 'panel-heading')}>
+				{headerItems}
+			</div>
+			<Panel
+				collapsible={!!content}
+				expanded={expanded}
+			>
+				{children}
+			</Panel>
+		</div>
+	);
 }
 
 CollapsiblePanel.propTypes = {
 	header: PropTypes.arrayOf(renderHeaderItem.propTypes).isRequired,
-	content: PropTypes.arrayOf(PropTypes.shape({
-		label: PropTypes.string,
-		description: PropTypes.string,
-	})),
+	onSelect: PropTypes.func,
+	onToggle: PropTypes.func,
+	expanded: PropTypes.bool,
+	selected: PropTypes.bool,
+	theme: PropTypes.string,
+	content: PropTypes.oneOf([
+		PropTypes.arrayOf(PropTypes.shape({
+			label: PropTypes.string,
+			description: PropTypes.string,
+		})),
+		PropTypes.shape({
+			upper: PropTypes.arrayOf(PropTypes.shape(simplePropTypes)),
```

Comment:
`upper` is not well chosen, I prefer `label` or `title` here, what do you think?